homeassistant/components/efergy/sensor.py

```diff
@@ -65,99 +99,99 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
+                        sid = sensor["sid"]
+                        dev.append(
+                            EfergySensor(
+                                app_token,
+                                utc_offset,
+                                variable[CONF_PERIOD],
+                                variable[CONF_CURRENCY],
+                                description,
+                                sid,
+                            )
+                        )
                 dev.append(
                     EfergySensor(
-                        variable[CONF_TYPE],
                         app_token,
                         utc_offset,
                         variable[CONF_PERIOD],
                         variable[CONF_CURRENCY],
-                        sid,
+                        description,
                     )
                 )
-        dev.append(
-            EfergySensor(
-                variable[CONF_TYPE],
-                app_token,
-                utc_offset,
-                variable[CONF_PERIOD],
-                variable[CONF_CURRENCY],
-            )
-        )
 
     add_entities(dev, True)
 
 
 class EfergySensor(SensorEntity):
     """Implementation of an Efergy sensor."""
 
-    def __init__(self, sensor_type, app_token, utc_offset, period, currency, sid=None):
+    def __init__(
+        self,
+        app_token,
+        utc_offset,
+        period,
+        currency,
+        description: SensorEntityDescription,
+        sid=None,
+    ):
         """Initialize the sensor."""
         self.sid = sid
         if sid:
-            self._name = f"efergy_{sid}"
+            self._attr_name = f"efergy_{sid}"
         else:
-            self._name = SENSOR_TYPES[sensor_type][0]
-        self.type = sensor_type
+            self._attr_name = description.name
+        self.type = description.key
         self.app_token = app_token
         self.utc_offset = utc_offset
-        self._state = None
         self.period = period
-        self.currency = currency
-        if self.type == "cost":
-            self._unit_of_measurement = f"{self.currency}/{self.period}"
+        if description.key == CONF_COST:
+            self._attr_native_unit_of_measurement = f"{currency}/{period}"
         else:
-            self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
-
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
-
-    @property
-    def native_value(self):
-        """Return the state of the sensor."""
-        return self._state
-
-    @property
-    def native_unit_of_measurement(self):
-        """Return the unit of measurement of this entity, if any."""
-        return self._unit_of_measurement
+            self._attr_native_unit_of_measurement = (
+                description.native_unit_of_measurement
+            )
+        self._attr_device_class = description.device_class
+        self._attr_state_class = description.state_class
```

Comment:
These can be removed, as they can be retrieved from the entity description automatically.